tests/test-transmuxer.c

```diff
@@ -102,6 +102,140 @@ test_corrupted (void)
+  g_autoptr (GDir) dir = NULL;
+  g_autoptr (GError) error = NULL;
+  const gchar *filename = NULL;
+  g_autofree gchar *filepath = NULL;
+
+  dir = g_dir_open (folder, 0, &error);
+  if (error)
+    return;
+
+  while ((filename = g_dir_read_name (dir))) {
+    filepath = g_build_filename (folder, filename, NULL);
+    g_unlink (filepath);
+  }
+
+  g_rmdir (folder);
+
+}
+
+static gint
+gchar_compare (gconstpointer a, gconstpointer b)
+{
+  gchar *str_a = *(gchar **) a;
+  gchar *str_b = *(gchar **) b;
+
+  return g_strcmp0 (str_a, str_b);
+}
+
+static void
+test_split_time_bytes (gint64 time, gint64 bytes)
+{
+  g_autoptr (HwangsaeTransmuxer) transmuxer = NULL;
+  g_autoptr (GError) error = NULL;
+  g_autofree gchar *output_folder = NULL;
+  g_autofree gchar *output_file = NULL;
+  GSList *input_files = NULL;
+  g_autoptr (GArray) file_list = NULL;
+  g_autoptr (GDir) dir = NULL;
+  const gchar *filename = NULL;
+
+  input_files = g_slist_append (input_files,
+      g_test_build_filename (G_TEST_DIST, "data/test-0-5000000.ts", NULL));
+  input_files = g_slist_append (input_files,
+      g_test_build_filename (G_TEST_DIST, "data/test-5000000-10000000.ts",
+          NULL));
+  input_files = g_slist_append (input_files,
+      g_test_build_filename (G_TEST_DIST, "data/test-10000000-15000000.ts",
+          NULL));
+
+  transmuxer = hwangsae_transmuxer_new ();
+
+  output_folder =
+      g_build_filename (g_get_tmp_dir (), "transmuxer-test-XXXXXX", NULL);
+  g_mkdtemp (output_folder);
+
+  g_debug ("using folder %s\n", output_folder);
+
+  output_file = g_build_filename (output_folder, "%05d.mp4", NULL);
+
+  if (time)
+    hwangsae_transmuxer_set_max_size_time (transmuxer, time);
+  else if (bytes)
+    hwangsae_transmuxer_set_max_size_bytes (transmuxer, bytes);
+
+  hwangsae_transmuxer_merge (transmuxer, input_files, output_file, &error);
+
+  g_assert_no_error (error);
+
+  g_slist_free_full (input_files, g_free);
+
+  // Check if the splitting is OK
+  dir = g_dir_open (output_folder, 0, &error);
+
+  g_assert_no_error (error);
+
+  file_list = g_array_new (TRUE, TRUE, sizeof (gchar *));
+  while ((filename = g_dir_read_name (dir))) {
+    g_array_append_val (file_list, filename);
+  }
+
+  g_array_sort (file_list, gchar_compare);
+
+  for (gint i = 0; i < file_list->len; i++) {
+    g_autofree gchar *filepath = NULL;
+
+    filename = g_array_index (file_list, gchar *, i);
+    filepath = g_build_filename (output_folder, filename, NULL);
+
+    g_debug ("Checking file %s", filename);
+    if (time) {
+      GstClockTime duration = hwangsae_test_get_file_duration (filepath);
+
+      g_debug ("%s has duration %" GST_TIME_FORMAT, filename,
+          GST_TIME_ARGS (duration));
+      if (i < file_list->len - 1) {
+        g_assert_cmpint (labs (GST_CLOCK_DIFF (duration, time)), <=,
+            GST_SECOND);
+      } else {
+        /* The final segment should be shorter than FILE_SEGMENT_LEN. */
+        g_assert_cmpint (duration, <, time);
```

Comment:
It may happen the last segment is exactly `time` long, which is still valid. In any case, the file must not be longer. Use `<=` for comparison.

Reply:
Sure, thanks.